cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -0,0 +1,114 @@
+"use client"
+import { deleteComment } from "@/app/actions/comments/deleteComment"
+import { setResolved } from "@/app/actions/comments/setResolved"
+import { upvoteComment } from "@/app/actions/comments/setLike"
+import { getUser } from "@/app/actions/users/getUser"
+import useSWR, { mutate } from "swr"
+
+export default function Comment({
+  comment,
+  userId,
+  childDepth = 0,
+}: {
+  comment: CommentModel
+  userId: string
+  childDepth?: number
+}) {
+  const date = new Date(comment.last_update)
+  const childOrParentStyling = childDepth
+    ? `p-3 ml-${6 / childDepth} lg:ml-${12 / childDepth} border-t border-gray-400 dark:border-gray-600`
+    : "p-6 mb-3 border-b border-gray-200 rounded-lg"
+
+  const deleteCommentAction = async () => {
+    await deleteComment(comment.id).finally(() => {
+      mutate(comment.document_id)
+    })
+  }
+
+  const setResolvedCommentAction = async (resolved: boolean) => {
+    await setResolved(comment.id, resolved).finally(() => {
+      mutate(comment.document_id)
+    })
+  }
+
+  const setLikedCommentAction = async (liked: boolean) => {
+    await upvoteComment(comment.id, liked, userId).finally(() => {
+      mutate(comment.document_id)
+    })
+  }
+
+  const { data: user } = useSWR(comment.user_id, getUser)
+  if (user && user.ok) {
```

Comment:
Refactor this.
1. return render in an if-else statement obscures readability
2. You have many conditional renders that would be more easily readable if separated, in a separate (local) component or as an object generated above.
3. Generally, early exit is preferred - if you want to present the error (or `return null`) in case `user` has not retrieved correctly, you should *start* with it, and do the normal return afterwards.